sqld/src/auth.rs

```diff
@@ -119,6 +120,40 @@ impl Auth {
     }
 }
 
+impl Authenticated {
+    pub fn from_proxy_grpc<T>(req: &tonic::Request<T>) -> Result<Self, Status> {
+        let auth = match req
+            .metadata()
+            .get(GRPC_PROXY_AUTH_HEADER)
+            .map(|v| v.to_str())
+            .transpose()
+            .map_err(|_| Status::invalid_argument("missing x-proxy-auth"))?
+        {
+            Some("full_access") => Authenticated::Authorized(Authorized::FullAccess),
+            Some("read_only") => Authenticated::Authorized(Authorized::ReadOnly),
+            Some("anonymous") => Authenticated::Anonymous,
+            Some(_) => return Err(Status::permission_denied("invalid authorization level")),
+            None => return Err(Status::invalid_argument("unable to convert to ascii")),
+        };
+
+        Ok(auth)
+    }
+
+    pub fn as_proxy_grpc<T>(&self, req: &mut tonic::Request<T>) {
```

Comment:
nit: The naming could be clearer. The `as_` has a strong connotation in rust, and I would expect to get something back (c.f https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). Maybe something like `update_gprc_request` would be a better fit?

Reply:
Yeah, I originally changed it because clippy yelled at me, will get a better name.